### PR TITLE
#163388212 Users should be able to login

### PR DESCRIPTION
**Pull request by bryan-munene**
#### What does this PR do?

This PR sets up the login feature for the application.

#### Description of Task to be completed?

- Add tests for this login feature
- Consume the login endpoint.

#### How should this be manually tested?

Access the feature on `https://ah-bird-box-frontend-staging.herokuapp.com/login`

#### Any background context you want to provide?

A user should be able to login to their account if they already have an account and provide the correct login credentials for their account.

#### What are the relevant pivotal tracker stories?

[#163388212](https://www.pivotaltracker.com/story/show/163388212)

#### Screenshots (if appropriate):

image

#### Questions:

None

#### Reviewers:

@TooColline
@calebrotich10
@unah254
@bryan-munene
@hoslack

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,8 +38,8 @@
     "not op_mini all"
   ],
   "devDependencies": {
-    "enzyme": "^3.9.0",
-    "enzyme-adapter-react-16": "^1.10.0",
+    "babel-eslint": "9.0.0",
+    "dotenv": "^6.2.0",
```
Comment (TooColline) [inline package.json]: Hey, @bryan-munene I suggest you get rid of `dotenv` because we don't need it.

**Review by calebrotich — APPROVED** [review]
Apart from the inline styling in Login.jsx component, everything else is well done.

File: src/components/Navigation.jsx [hunk 2]
```diff
@@ -1,19 +1,52 @@
...
+// eslint-disable-next-line react/prefer-stateless-function
+class NavigationView extends Component {
+  // Create a Navigation view of the application
+  logout = () => {
+    localStorage.removeItem('user');
+    localStorage.removeItem('token');
+    window.location.assign('/');
+  }
 
+  // isAuth() {
```
Comment (TooColline) [inline src/components/Navigation.jsx]: Hey @bryan-munene please remove the commented out code

**Review by unah254 — COMMENTED** [review]
Apart from the inline comments as mentioned by Collins, looks good to me

File: src/config/configAxios.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+const token = localStorage.getItem('token');
+const axiosConfig = axios;
+if (token) {
+  axios.defaults.headers.common = {
+    'Content-Type': 'application/json',
+    Authorization: `Bearer ${token}`,
+  };
+}
+axios.defaults.baseURL = 'https://ah-bird-box-staging.herokuapp.com/api';
```
Comment (hoslack) [inline src/config/configAxios.js]: Hide this in an ENV

**Review by hoslack — CHANGES_REQUESTED** [review]

File: src/config/configAxios.js (hunk 3 above)
Comment (bryan-munene) [reply src/config/configAxios.js]: done

**Review by hoslack — APPROVED** [review]

**Pull request merged**